src/Illuminate/Database/Query/Builder.php

```diff
@@ -1420,17 +1420,27 @@ public function simplePaginate($perPage = 15, $columns = ['*'])
 	/**
 	 * Get the count of the total records for the paginator.
 	 *
+	 * @param  array  $columns
 	 * @return int
 	 */
-	public function getCountForPagination()
+	public function getCountForPagination($columns = ['*'])
 	{
 		$this->backupFieldsForCount();
 
-		$total = $this->count();
+		$this->aggregate = ['function' => 'count', 'columns' => $columns];
+
+		$results = $this->get();
+
+		$this->aggregate = null;
 
 		$this->restoreFieldsForCount();
 
-		return $total;
+		if (isset($results[0]))
+		{
+			if (isset($this->groups)) return count($results);
+
+			return (int) array_change_key_case((array) $results[0])['aggregate'];
```

Comment:
Ok, good. I couldn't remember if it was added in 5.4, or 5.5. :) I only use 5.5/5.6 on prod.